### PR TITLE
Generalize .gitignore

### PR DESCRIPTION
**Pull request by dominichofer**
`*.o` is moved.
`*_mod.mod` and `class_gribfile.mod` are generalized by `*.mod`

### CODE REVIEW

**Comment by dominichofer** [comment]
@pirmink Are you interested in this extension? If so, please review and merge. I don't have the right to assign reviewers.

**Comment by pirmink** [comment]
Ok, but where did you *.o and *.mod move to? I only see one .gitignore file at root level changed by this PR, with *.o and *.mod removed.

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,41 @@
...
 FLEXPART
 FLEXPART_MPI
+
+# Prerequisites
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
```
Comment (dominichofer) [inline .gitignore:13]: here's `*.o`

File: .gitignore [hunk 2]
```diff
@@ -1,8 +1,41 @@
...
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Fortran module files
+*.mod
```
Comment (dominichofer) [inline .gitignore:26]: here's `*.mod`

Comment (pirmink) [reply .gitignore:26]: Ah, yes, I only looked at the red lines, sorry. All good!

**Pull request merged**